### PR TITLE
BACKLOG-40268:Servlet for step actions

### PR DESCRIPTION
**Pull request by addagudi**

### CODE REVIEW

File: plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java [hunk 1]
```diff
@@ -142,4 +145,55 @@ public Object normalizeValue( ValueMetaInterface valueMeta, Object value ) throw
...
+      String realUsername = getTransMeta().environmentSubstitute( meta.getUsername() );
+      String realPassword = Utils.resolvePassword( getTransMeta(), meta.getPassword() );
+      int realTimeOut = Const.toInt( getTransMeta().environmentSubstitute( meta.getTimeout() ), 0 );
+
+      connection = new SalesforceConnection( log, realURL, realUsername, realPassword );
+      connection.setTimeOut( realTimeOut );
+      connection.connect();
+
+    } catch ( Exception e ) {
+      successConnection = false;
```
Comment (sVarma1992) [inline plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java:187]: better to log error here

Comment (sVarma1992) [inline plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java:187]: we are not throwing the exception here so in the dialog class the is no point of catching and writing some code inside catch block

Comment (addagudi) [reply plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java:187]: Done

Comment (addagudi) [reply plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java:187]: We are not catching this exception in dialog. The try catch there is for a creating an instance of Meta class and it's needed.

File: plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java [hunk 2]
```diff
@@ -142,4 +145,55 @@ public Object normalizeValue( ValueMetaInterface valueMeta, Object value ) throw
...
+    } finally {
+      if ( connection != null ) {
+        try {
+          connection.close();
+        } catch ( Exception e ) {
+          successConnection = false;
```
[the remark this answers is not in the document]
Comment (addagudi) [reply plugins/salesforce/core/src/main/java/org/pentaho/di/trans/steps/salesforce/SalesforceStep.java:194]: done

**Review by sVarma1992 — APPROVED** [review]

**Review by hv-vamsi — APPROVED** [review]

**Pull request merged**